planner/core/runtime_filter_generator.go

```diff
@@ -0,0 +1,233 @@
+				zap.Int32("PlanNodeId", int32(hashJoinPlan.id)),
+				zap.String("JoinType", hashJoinPlan.JoinType.String()))
+			return false
+		}
+	} else {
+		// case2: build side is on the left
+		if hashJoinPlan.JoinType == RightOuterJoin {
+			logutil.BgLogger().Debug("Join type does not match RF pattern when build side is on the left",
+				zap.Int32("PlanNodeId", int32(hashJoinPlan.id)),
+				zap.String("JoinType", hashJoinPlan.JoinType.String()))
+			return false
+		}
+	}
+	return true
+}
+
+func (generator *RuntimeFilterGenerator) matchEQPredicate(eqPredicate *expression.ScalarFunction,
+	rightIsBuildSide bool) bool {
+	var targetColumn, srcColumn *expression.Column
+	if rightIsBuildSide {
+		targetColumn = eqPredicate.GetArgs()[0].(*expression.Column)
+		srcColumn = eqPredicate.GetArgs()[1].(*expression.Column)
+	} else {
+		targetColumn = eqPredicate.GetArgs()[1].(*expression.Column)
+		srcColumn = eqPredicate.GetArgs()[0].(*expression.Column)
+	}
+	// match target column
+	// condition1: the target column must be real column
+	// condition2: the target column has not undergone any transformation
+	// todo: cast expr
+	if targetColumn.IsHidden || targetColumn.OrigName == "" {
+		logutil.BgLogger().Debug("Target column does not match RF pattern",
+			zap.String("EQPredicate", eqPredicate.String()),
+			zap.String("TargetColumn", targetColumn.String()),
+			zap.Bool("IsHidden", targetColumn.IsHidden),
+			zap.String("OrigName", targetColumn.OrigName))
+		return false
+	}
+	// match data type
+	srcColumnType := srcColumn.GetType().GetType()
+	if srcColumnType == mysql.TypeJSON || srcColumnType == mysql.TypeBlob ||
+		srcColumnType == mysql.TypeLongBlob || srcColumnType == mysql.TypeMediumBlob ||
+		srcColumnType == mysql.TypeTinyBlob || srcColumn.GetType().Hybrid() || srcColumn.GetType().IsArray() {
+		logutil.BgLogger().Debug("Src column type does not match RF pattern",
+			zap.String("EQPredicate", eqPredicate.String()),
+			zap.String("SrcColumn", srcColumn.String()),
+			zap.String("SrcColumnType", srcColumn.GetType().String()))
+		return false
+	}
+	return true
+}
+
+func (generator *RuntimeFilterGenerator) calculateRFMode(buildNode *PhysicalHashJoin, targetNode *PhysicalTableScan) variable.RuntimeFilterMode {
+	if generator.belongsToSameFragment(buildNode, targetNode) {
+		return variable.Local
+	} else {
+		return variable.Global
+	}
+}
+
+func (generator *RuntimeFilterGenerator) belongsToSameFragment(currentNode PhysicalPlan, targetNode *PhysicalTableScan) bool {
+	switch currentNode.(type) {
+	case *PhysicalExchangeReceiver:
+		// terminal traversal
+		return false
+	case *PhysicalTableScan:
+		if currentNode.ID() == targetNode.id {
+			return true
+		}
+		return false
+	default:
+		for _, childNode := range currentNode.Children() {
+			if generator.belongsToSameFragment(childNode, targetNode) {
+				return true
+			}
+		}
+		return false
+	}
+}
```

Comment:
This func is not sufficient enough. We can move this to the place we cut fragments?

Reply:
If the target node and source node are belongs to different fragment, the RF mode should be global. Now, we need to judge if RF is global or local. If RF is global, we doesn't support now, so we need to remove it.
So it cannot be executed in the place where the fragment is split later.